From 5e5b8e1540153ed8fc997cc38d766d19496c0c0f Mon Sep 17 00:00:00 2001
From: Stephan Hartmann <stha09@googlemail.com>
Date: Thu, 28 Sep 2023 14:35:35 +0000
Subject: libstdc++: fix incomplete type in ui::GestureConsumer

libstdc++ uses sizeof in the destructor. Therefore, it needs the
full definition of GestureProviderAura.

Bug: 957519
Change-Id: I2c5b20892726f470a48fd307da430dc9b94b2b29
Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/4892401
Auto-Submit: Stephan Hartmann <stha09@googlemail.com>
Reviewed-by: Robert Flack <flackr@chromium.org>
Commit-Queue: Robert Flack <flackr@chromium.org>
Cr-Commit-Position: refs/heads/main@{#1202605}
---
 ui/events/gestures/gesture_types.cc | 4 ++++
 ui/events/gestures/gesture_types.h  | 4 +---
 2 files changed, 5 insertions(+), 3 deletions(-)

diff --git a/ui/events/gestures/gesture_types.cc b/ui/events/gestures/gesture_types.cc
index d615d2ca4d906..b0cd211969d70 100644
--- a/ui/events/gestures/gesture_types.cc
+++ b/ui/events/gestures/gesture_types.cc
@@ -23,6 +23,10 @@ base::WeakPtr<GestureConsumer> GestureConsumer::GetWeakPtr() {
   return weak_ptr_factory_.GetWeakPtr();
 }
 
+std::unique_ptr<GestureProviderAura> GestureConsumer::TakeProvider() {
+  return std::move(provider_);
+}
+
 void GestureConsumer::reset_gesture_provider() {
   provider_.reset();
 }
diff --git a/ui/events/gestures/gesture_types.h b/ui/events/gestures/gesture_types.h
index e7085424d1c64..e6920030f5adc 100644
--- a/ui/events/gestures/gesture_types.h
+++ b/ui/events/gestures/gesture_types.h
@@ -47,9 +47,7 @@ class EVENTS_EXPORT GestureConsumer {
 
   base::WeakPtr<GestureConsumer> GetWeakPtr();
 
-  std::unique_ptr<GestureProviderAura> TakeProvider() {
-    return std::move(provider_);
-  }
+  std::unique_ptr<GestureProviderAura> TakeProvider();
   void reset_gesture_provider();
   void set_gesture_provider(std::unique_ptr<GestureProviderAura> provider);
   GestureProviderAura* provider() const { return provider_.get(); }
-- 
2.39.2

